c++/src/kj/compat/http.c++

```diff
@@ -1113,11 +1113,63 @@ kj::String HttpHeaders::toString() const {
 
 namespace {
 
+template <typename Subclass>
+class WrappableStreamMixin {
+  // Both HttpInputStreamImpl and HttpOutputStream are commonly wrapped by a class that implements
+  // a particular type of body stream, such as a chunked body or a fixed-length body. That wrapper
+  // stream is passed back to the application to represent the specific request/response body, but
+  // the inner stream is associated with the connection and can be reused several times.
+  //
+  // It's easy for applications to screw up and hold on to a body stream beyond the lifetime of the
+  // underlying connection stream. This used to lead to UAF. This mixin class implements behavior
+  // that detached the wrapper if it outlives the wrapped stream, so that we log errors and
+
+public:
+  ~WrappableStreamMixin() noexcept(false) {
+    KJ_IF_MAYBE(w, currentWrapper) {
+      KJ_LOG(ERROR, "HTTP connection destroyed while HTTP body streams still exist",
+          kj::getStackTrace());
```

Comment:
Do we still need the log? I added it in my version so we could investigate further but not sure it's something we want to keep around indefinitely?

Reply:
I think so, it's definitely an error on the part of the app which we'd want to know about and fix. Actually it should probably be an exception in debug mode...